docs/visualization.rst

```diff
@@ -154,6 +161,34 @@ It also enables the coordinates along the plot axes to be updated between frames
   >>> plt.cla()
   >>> plt.close()
 
+Adding Colorbars
+----------------
+Working with the output of `ndcube.NDCube.plot` is the context of matplotlib figures and axes can be a great way of creating more complex plots.
+Here we will show two examples of home to add a colorbar.
+
+The first is simple and depends on `matplotlib.pyplot`.
+
+.. code-block:: python
+
+  >>> my_cube.plot()  # doctest: +SKIP
+  >>> plt.colorbar()  # doctest: +SKIP
+
+The second example shows how to more intricately play with `~ndcube.NDCube` visualizations and matplotlib figures and axes.
+This includes adding the output of `ndcube.NDCube.plot` to an existing axes object.
+
+.. plot::
+  :context:
+  :include-source:
+
+  >>> fig = plt.figure()  # Create a figure
+  >>> # Create WCSAxes object and then add the NDCube plot by setting the axes kwarg.
+  >>> ax = fig.add_axes([0.1, 0.1, 0.6, 0.6], projection=my_cube[0].wcs)
+  >>> ax = my_cube[0].plot(axes=ax)
+  >>> # Create the colorbar axes object and scale it by the image.
+  >>> cax = fig.add_axes([0.85, 0.1, 0.05, 0.6])
+  >>> im = ax.get_images()[0]  # Retrieve the plot AxesImage by which to scale colorbar.
+  >>> cbar = fig.colorbar(im, cax=cax, label="Intensity")
```

Comment:
I think this should work:

```suggestion
  >>> fig = plt.figure()  # Create a figure
  >>> # Create WCSAxes object and then add the NDCube plot by setting the axes kwarg.
  >>> ax = plt.subplot(projection=my_cube[0])
  >>> ax = my_cube[0].plot(axes=ax)
  >>> cbar = plt.colorbar(label="Intensity")
```

Reply:
even the `(axes=ax)` is optional.

Reply:
I like leaving in `axes=ax`, as it's explicit and shows the user how to choose which axes to plot their cube on.

Reply:
Does `fig.subplot` work instead? Again, then it's more explicit.

Reply:
I've tried to make the above example as short and easy as possible and the second to me much more explicit.

Reply:
I always feel like `fig.add_axes` is too much for 99% of the time. Why on earth do you want to be manually defining the size of your axes like this? If you really care use gridspec or something.

We could show an example using only the mpl OO API though, if `fig.subplot` does indeed work.

Reply:
I'm going to leave that for someone who knows about this stuff to make in a separate PR.